### ISSUE TITLE
Should I continue to use gradient clipping?

### ISSUE THREAD

**Issue by Rhuax** [NVIDIA/apex#201]
I'm applying amp to an existing model where before the loss.backward() I was applying some gradient clipping with torch.nn.utils.clip_grad_norm_(). Should I keep gradient clipping after applying amp?

**Comment by mcarilli** [comment]
If gradient clipping was useful to you without Amp (pure FP32 training) then you can and should continue to use it.  However, it requires some special syntax:
https://nvidia.github.io/apex/advanced.html#gradient-clipping
